ui-components/src/actions/FavoriteButton.tsx

```diff
@@ -0,0 +1,114 @@
+import * as React from "react"
+import {
+  AppearanceSizeType,
+  AuthContext,
+  Button,
+  Icon,
+  IconFillColors,
+} from "@bloom-housing/ui-components"
+import { t } from "../helpers/translator"
+import { useContext, useEffect, useState } from "react"
+
+export interface FavoriteButtonProps {
+  removeFavoriteOnSuccess?: () => void
+  addFavoriteOnSuccess?: () => void
+  id: string
+  allowFavoriting?: boolean
+}
+
+const FavoriteButton = ({
+  removeFavoriteOnSuccess,
+  addFavoriteOnSuccess,
```

Comment:
It looks like these aren't passed in anywhere and we can remove the need for them, unless there is a valid reason to keep them now.